### PR TITLE
Add suport for Multichoice fields and filters

### PR DESCRIPTION
**Pull request by maurizi**
To test:
 - Change the API reference in `OpenTreeMapSkined/res/values/overrides.xml` from v3 to v4
 - Use https://github.com/OpenTreeMap/otm-core/pull/2465 and https://github.com/OpenTreeMap/otm-addons/pull/1054 to add a multichoice field to your mobile search fields and mobile detail fields
 - You should be able to add a tree and set its multichoice field, and search by multichoice fields

### CODE REVIEW

File: OpenTreeMap/src/main/java/org/azavea/otm/fields/MultiChoiceField.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    checkedChoices[valueIndex] = true;
+                }
+            }
+
+            String[] choices = editedField.choiceDisplayValues.toArray(
+                    new String[editedField.choiceDisplayValues.size()]);
+
+            new AlertDialog.Builder(choiceButton.getContext())
+                    .setTitle(editedField.label)
+                    .setPositiveButton("Ok", (dialog, which) -> {
```
Comment (maurizi) [inline OpenTreeMap/src/main/java/org/azavea/otm/fields/MultiChoiceField.java]: Note to self: need to extract this as a string resource

**Comment by steventlamb** [comment]
Getting app crashes repeatably when I edit from the plot detail page within the app. It looks like I am only able to produce when moving to and from null/empty values in multichoice fields. Two crashes appear in logs, do not know which is crashing the app:
#606 java.lang.ClassCastException: org.json.JSONObject$1 cannot be cast to org.json.JSONArray
#607 java.lang.NullPointerException: Unable to render tree view

I'm assuming it's the latter, else how would the latter happen.

**Comment by steventlamb** [comment]
The previous issue appeas to be fixed.

**Comment by steventlamb** [comment]
I'm getting tile render errors when I attempt to search *or* filter by a multichoice field.
```
[TILE RINDER ERROR] -- 400: Error: The model name must be one of the following: mapFeature, polygonalMapFeature, tree, plot, rainBarrel, rainGarden, bioswale, species, mapFeaturePhoto, udf. Not 0
```

**Comment by jwalgran** [comment]
I was seeing these "Not 0" errors when testing the iOS app, too, and I saw them without any multichoice fields present. I did some debugging and opened a PR on the tiler.

OpenTreeMap/otm-tiler#103

**Comment by jwalgran** [comment]
OpenTreeMap/otm-tiler#103 is merged. If you update your tiler you should be able to test cleanly.

**Comment by steventlamb** [comment]
+1

**Comment by steventlamb** [comment]
tested

**Pull request merged**